consumer/spring-websocket-consumer/src/main/java/org/springframework/cloud/fn/consumer/websocket/trace/InMemoryTraceRepository.java

```diff
@@ -41,35 +43,50 @@ public class InMemoryTraceRepository {
 
 	private final List<Trace> traces = new LinkedList<>();
 
+	private final Lock tracesLock = new ReentrantLock();
+
 	/**
 	 * Flag to say that the repository lists traces in reverse order.
 	 * @param reverse flag value (default true)
 	 */
 	public void setReverse(boolean reverse) {
-		synchronized (this.traces) {
+		try {
+			this.tracesLock.lock();
 			this.reverse = reverse;
 		}
+		finally {
+			this.tracesLock.unlock();
+		}
 	}
 
 	/**
 	 * Set the capacity of the in-memory repository.
 	 * @param capacity the capacity
 	 */
 	public void setCapacity(int capacity) {
-		synchronized (this.traces) {
+		try {
+			this.tracesLock.lock();
 			this.capacity = capacity;
 		}
+		finally {
+			this.tracesLock.unlock();
+		}
 	}
 
 	public List<Trace> findAll() {
-		synchronized (this.traces) {
+		try {
+			this.tracesLock.lock();
```

Comment:
Doesn't look like this operation needs any locking at all.

Reply:
Alright, then I’m removing the synchronized keyword as well. Is it correct?

Reply:
@artembilan I think the same way here as well; we shouldn’t remove the locking from this part either.